Framework/Geometry/src/Instrument.cpp

```diff
@@ -1338,6 +1345,28 @@ boost::shared_ptr<ParameterMap> Instrument::makeLegacyParameterMap() const {
   return pmap;
 }
 
+/** Parse the instrument tree and create ComponentInfo and DetectorInfo.
+ *
+ * This can be called for the base instrument once it is completely created, in
+ * particular when it is stored in the InstrumentDataService for reusing it
+ * later and avoiding repeated tree walks if several workspaces with the same
+ * instrument are loaded. */
+void Instrument::parseTree() {
+  if (isParametrized())
+    throw std::logic_error("Instrument::parseTree must be called with the base "
+                           "instrument, not a parametrized instrument");
+  std::tie(m_componentInfo, m_detectorInfo) =
+      InstrumentVisitor::makeWrappers(*this);
+}
+
+const ComponentInfo *Instrument::componentInfo() const {
```

Comment:
This method is not acting as an object source, and should probably be commented as such. Maybe consider making the pointer itself const too  to avoid accidental deletes?